lib/helpers/setDefaultsOnInsert.js

```diff
@@ -113,5 +115,13 @@ function isModified(modified, path) {
     }
     cur += '.' + sp[i];
   }
+
+  // Is any child of `path` modified?
+  for (const modifiedPath of Object.keys(modified)) {
+    if (modifiedPath.slice(0, path.length + 1) === path + '.') {
```

Comment:
I wonder if 

```js
const rootPath = path + '.'

for (const modifiedPath of Object.keys(modified)) {
  If (modifiedPath.startsWith(rootPath)) {
    return true;
  }
}
```

We would create less intermediate strings, so i assume better perf.